### PR TITLE
pod.yaml: add init container waiting for free port

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: bindata/v3.11.0/kube-apiserver/pod.yaml [hunk 1]
```diff
@@ -13,10 +13,21 @@ spec:
     - name: fix-audit-permissions
       image: ${IMAGE}
       imagePullPolicy: IfNotPresent
-      command: ['sh', '-c', 'chmod 0700 /var/log/kube-apiserver']
+      command: ['/bin/sh', '-c', 'chmod 0700 /var/log/kube-apiserver']
       volumeMounts:
         - mountPath: /var/log/kube-apiserver
           name: audit-dir
+    - name: wait-for-port
```
Comment (mfojtik) [inline bindata/v3.11.0/kube-apiserver/pod.yaml]: nit: wait-for-host-port

File: bindata/v3.11.0/kube-apiserver/pod.yaml [hunk 2]
```diff
@@ -13,10 +13,21 @@ spec:
...
+      image: ${IMAGE}
+      imagePullPolicy: IfNotPresent
+      command: ['/usr/bin/timeout', '90', "/bin/bash", "-c"]
+      args:
+      - |
+        echo -n "Waiting for port :6443 to be released."
```
Comment (mfojtik) [inline bindata/v3.11.0/kube-apiserver/pod.yaml]: remember that BZ that was asking us to make this port configurable?

File: bindata/v3.11.0/kube-apiserver/pod.yaml [hunk 3]
```diff
@@ -13,10 +13,21 @@ spec:
...
       imagePullPolicy: IfNotPresent
-      command: ['sh', '-c', 'chmod 0700 /var/log/kube-apiserver']
+      command: ['/bin/sh', '-c', 'chmod 0700 /var/log/kube-apiserver']
       volumeMounts:
         - mountPath: /var/log/kube-apiserver
           name: audit-dir
+    - name: wait-for-port
+      image: ${IMAGE}
+      imagePullPolicy: IfNotPresent
+      command: ['/usr/bin/timeout', '90', "/bin/bash", "-c"]
```
Comment (mfojtik) [inline bindata/v3.11.0/kube-apiserver/pod.yaml]: does this correlate with the time revision controller is willing to wait for apiserver to be up?

File: bindata/v3.11.0/kube-apiserver/pod.yaml [hunk 4]
```diff
@@ -13,10 +13,21 @@ spec:
...
+      args:
+      - |
+        echo -n "Waiting for port :6443 to be released."
+        while [ -n "$(lsof -ni :6443)" ]; do
+          echo -n "."
```
Comment (mfojtik) [inline bindata/v3.11.0/kube-apiserver/pod.yaml]: 90 dots in my log :-)

File: bindata/v3.11.0/kube-apiserver/pod.yaml [hunk 5]
```diff
@@ -13,10 +13,21 @@ spec:
...
         - mountPath: /var/log/kube-apiserver
           name: audit-dir
+    - name: wait-for-port
+      image: ${IMAGE}
+      imagePullPolicy: IfNotPresent
+      command: ['/usr/bin/timeout', '90', "/bin/bash", "-c"]
+      args:
+      - |
+        echo -n "Waiting for port :6443 to be released."
+        while [ -n "$(lsof -ni :6443)" ]; do
```
Comment (mfojtik) [inline bindata/v3.11.0/kube-apiserver/pod.yaml]: thought we talked about `netstat`

File: bindata/v3.11.0/kube-apiserver/pod.yaml (hunk 3 above)
Comment (sttts) [reply bindata/v3.11.0/kube-apiserver/pod.yaml]: 90 >> 60 seconds for graceful termination. But I wonder whether the 34 sec extra wait time must be added.

File: bindata/v3.11.0/kube-apiserver/pod.yaml (hunk 2 above)
Comment (sttts) [reply bindata/v3.11.0/kube-apiserver/pod.yaml]: When we implement that, we make this more generic here too. Probably the whole thing will turn into a template with port variable.

File: bindata/v3.11.0/kube-apiserver/pod.yaml (hunk 5 above)
Comment (sttts) [reply bindata/v3.11.0/kube-apiserver/pod.yaml]: netstat is old-school (I believe, @jeremyeder can probably estimate your age just by your question here ;-) )

File: bindata/v3.11.0/kube-apiserver/pod.yaml (hunk 4 above)
Comment (sttts) [reply bindata/v3.11.0/kube-apiserver/pod.yaml]: 90 should be enough for everybody. I can put in a newline after 72 if you like ;-)

File: bindata/v3.11.0/kube-apiserver/pod.yaml (hunk 3 above)
Comment (sttts) [reply bindata/v3.11.0/kube-apiserver/pod.yaml]: updated to 100: 60 for graceful termination, 35 for https://github.com/openshift/cluster-kube-apiserver-operator/pull/318

**Pull request merged**